controller/kv/helper.go

```diff
@@ -2872,40 +2883,145 @@ func (m clusterHelper) PutImportTask(importTask *share.CLUSImportTask) error {
-    return apikeys
+	apikeys := make(map[string]*share.CLUSApikey)
+	keys, _ := cluster.GetStoreKeys(share.CLUSConfigApikeyStore)
+	for _, key := range keys {
+		if value, _, _ := m.get(key); value != nil {
+			var apikey share.CLUSApikey
+			json.Unmarshal(value, &apikey)
+			apikeys[apikey.Name] = &apikey
+		}
+	}
+	return apikeys
 }
 
 func (m clusterHelper) DeleteApikey(name string) error {
 	key := share.CLUSApikeyKey(url.QueryEscape(name))
 	return cluster.Delete(key)
 }
+
+// sigstore
+func (m clusterHelper) PutSigstoreRootOfTrust(rootOfTrust *share.CLUSSigstoreRootOfTrust) error {
+	value, err := enc.Marshal(rootOfTrust)
+	if err != nil {
+		return err
+	}
+	return cluster.Put(share.CLUSSigstoreRootOfTrustKey(rootOfTrust.Name), value)
+}
+
+func (m clusterHelper) GetSigstoreRootOfTrust(rootName string) (*share.CLUSSigstoreRootOfTrust, error) {
+	rootOfTrustData, _, err := m.get(share.CLUSSigstoreRootOfTrustKey(rootName))
+	if err != nil || rootOfTrustData == nil {
+		return nil, err
+	}
+	rootOfTrust := &share.CLUSSigstoreRootOfTrust{}
+	err = json.Unmarshal(rootOfTrustData, rootOfTrust)
+	if err != nil {
+		return nil, err
+	}
+	return rootOfTrust, nil
+}
+
+func (m clusterHelper) DeleteSigstoreRootOfTrust(rootName string) error {
+	return cluster.Delete(share.CLUSSigstoreRootOfTrustKey(rootName))
+}
+
+func (m clusterHelper) GetAllSigstoreRootsOfTrust() (rootOfTrust map[string]*share.CLUSSigstoreRootOfTrust, err error) {
+	keys, err := cluster.GetStoreKeys(share.CLUSConfigSigstoreRootsOfTrust)
+	if err != nil && err.Error() != "Empty store" {
+		return nil, err
+	}
+	rootsOfTrust := make(map[string]*share.CLUSSigstoreRootOfTrust, len(keys))
+	for _, key := range keys {
+		if value, _, err := m.get(key); value != nil {
+			if err != nil {
+				return nil, fmt.Errorf("could not retrieve all roots of trust, error retrieving \"%s\": %s", key, err.Error())
+			}
+			rootsOfTrust[key] = &share.CLUSSigstoreRootOfTrust{}
+			err = json.Unmarshal(value, rootsOfTrust[key])
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	return rootsOfTrust, nil
+}
+
+func (m clusterHelper) PutSigstoreVerifier(rootName string, verifier *share.CLUSSigstoreVerifier) error {
+	if !cluster.Exist(share.CLUSSigstoreRootOfTrustKey(rootName)) {
+		return fmt.Errorf("root \"%s\" does not exist", rootName)
+	}
+	value, err := enc.Marshal(verifier)
+	if err != nil {
+		return err
+	}
+	return cluster.Put(share.CLUSSigstoreVerifierKey(rootName, verifier.Name), value)
+}
+
+func (m clusterHelper) GetSigstoreVerifier(rootName string, verifierName string) (*share.CLUSSigstoreVerifier, error) {
+	if !cluster.Exist(share.CLUSSigstoreRootOfTrustKey(rootName)) {
+		return nil, fmt.Errorf("root \"%s\" does not exist", rootName)
+	}
+	verifierData, _, err := m.get(share.CLUSSigstoreVerifierKey(rootName, verifierName))
+	if err != nil || verifierData == nil {
+		return nil, err
+	}
+	verifier := &share.CLUSSigstoreVerifier{}
+	err = json.Unmarshal(verifierData, verifier)
```

Comment:
in line 2975 enc.Marshal() is called but here json.Unmarshal() is called & it will always get incorrect json object.